lib/ball.js

```diff
@@ -23,50 +23,57 @@ let ball = class Ball {
     this.x += this.xSpeed;
     this.y += this.ySpeed;
     return this;
-  };
+  }
 
   wallCollision(canvas) {
-    //left or right
-    if(this.x + this.xSpeed > canvas.width - this.radius ||
-       this.x + this.xSpeed < 0 + this.radius) {
+    //left
```

Comment:
So it looks like you're just calculating whether the ball will end up in the wall before it gets there instead of seeing if it's colliding? is that the difference basically?